### PR TITLE
ISREST-1006: add sepa component

### PR DESCRIPTION
**Pull request by skoch-intershop**
## PR Type

 [x] Feature  

requires **CONCARDIS 1.7.2** release ( and an ICM release 7.10.16.6+ )

## What Is the Current Behavior?

It is not possible to create a Concardis SEPA Direct Debit payment instrument on the checkout payment page.
Issue Number: ISREST-1006

## What Is the New Behavior?
Creation of a Concardis SEPA Direct Debit payment instrument is possible on the checkout payment page.

## Does this PR Introduce a Breaking Change?

 [ ] Yes  
 [x] No

### CODE REVIEW

File: src/app/pages/checkout-payment/payment-concardis-directdebit/payment-concardis-directdebit.component.ts [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+      this.errorMessage.bic.message = undefined;
+      this.errorMessage.bic.messageKey = undefined;
+      this.errorMessage.bic.code = undefined;
+    }
+  }
+
+  /**
+   * determine errorMessages on the basis of the error code
+   */
+  getErrorMessage(code: number, fieldType: string, defaultMessage: string): string {
```
Comment (SGrueber) [inline src/app/pages/checkout-payment/payment-concardis-directdebit/payment-concardis-directdebit.component.ts]: move to parent component

File: src/app/pages/checkout-payment/payment-concardis-directdebit/payment-concardis-directdebit.component.ts [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+
+    return messageKey;
+  }
+
+  /* ---------------------------------------- cancel and submit form  ------------------------------------------- */
+
+  /**
+   * cancel new payment instrument, hides and resets the parameter form
+   */
+  cancelNewPaymentInstrument() {
```
Comment (SGrueber) [inline src/app/pages/checkout-payment/payment-concardis-directdebit/payment-concardis-directdebit.component.ts]: move to parent component

File: src/app/shared/forms-dynamic/components/checkbox-dynamic/checkbox-dynamic.component.html [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<div *ngIf="field.fieldGroupClassName" class="{{ field.fieldGroupClassName }}">
+  <ish-checkbox
```
Comment (SGrueber) [inline src/app/shared/forms-dynamic/components/checkbox-dynamic/checkbox-dynamic.component.html]: ```<div class="form-group"``` is missing

File: src/assets/i18n/en_US.json [hunk 4]
```diff
@@ -2679,5 +2679,17 @@
   "quote.already_in_basket.error": "The quote is already in the cart.",
   "quote.not_found.error": "The quote could not be found.",
   "basket.add_quote.error": "The quote could not be added to the cart.",
-  "quoterequest.not_editable.error": "You have already submitted the quote request. Please <a href=\"javascript:location.reload()\">reload</a> this page to view the changes."
+  "quoterequest.not_editable.error": "You have already submitted the quote request. Please <a href=\"javascript:location.reload()\">reload</a> this page to view the changes.",
+  "checkout.sepa.iban.error.default": "Field is required and should not be empty!",
```
Comment (SGrueber) [inline src/assets/i18n/en_US.json]: pls, use existing error message keys, if possible, e.g. checkout.payment.debit.transfer.account.number.missing.error
if it is not possible for some keys, ask docu team for assistence and provide also German and French translations

**Comment by shauke** [comment]
Changes cherry picked to 0.18.1 hotfix release branch (https://github.com/intershop/intershop-pwa/tree/chore/release_0.18.1). Will be released and merged via the hotfix branch.

**Comment by shauke** [comment]
Merged with 0.18.1 hotfix release changes in the develop branch (https://github.com/intershop/intershop-pwa/commit/c0db22434c5c242c5d0f22d5868e575dd5fa00d3).

**Pull request closed without merging**